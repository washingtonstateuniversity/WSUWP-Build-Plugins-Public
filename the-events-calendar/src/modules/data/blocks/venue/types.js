/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/VENUE_BLOCK/SET_INITIAL_STATE`;

export const SET_VENUE = `${ PREFIX_EVENTS_STORE }/SET_VENUE`;

export const TOGGLE_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP`;
export const SET_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP`;

export const TOGGLE_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP_LINK`;
export const SET_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP_LINK`;
