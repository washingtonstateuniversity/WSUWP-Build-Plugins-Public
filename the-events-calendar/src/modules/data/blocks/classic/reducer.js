/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
