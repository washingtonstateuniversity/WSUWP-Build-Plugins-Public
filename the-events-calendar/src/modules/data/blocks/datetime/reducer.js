/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
