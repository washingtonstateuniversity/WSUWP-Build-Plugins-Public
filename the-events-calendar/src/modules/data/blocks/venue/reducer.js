/**
 * Internal dependencies
 */
import * as types from './types';
import { editorDefaults } from '@moderntribe/common/utils/globals';

export const DEFAULT_STATE = {
	venue: editorDefaults().venue ? editorDefaults().venue : 0,
	showMap: true,
	showMapLink: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_VENUE:
			return {
				...state,
				venue: action.payload.venue,
			};
		case types.TOGGLE_VENUE_MAP:
			return {
				...state,
				showMap: ! state.showMap,
			};
		case types.SET_VENUE_MAP:
			return {
				...state,
				showMap: action.payload.showMap,
			};
		case types.TOGGLE_VENUE_MAP_LINK:
			return {
				...state,
				showMapLink: ! state.showMapLink,
			};
		case types.SET_VENUE_MAP_LINK:
			return {
				...state,
				showMapLink: action.payload.showMapLink,
			};
		default:
			return state;
	}
};
