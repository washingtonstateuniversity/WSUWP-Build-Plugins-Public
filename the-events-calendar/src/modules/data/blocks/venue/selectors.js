/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const venueBlockSelector = ( state ) => state.events.blocks.venue;
export const getVenue = createSelector(
	[ venueBlockSelector ],
	( block ) => block.venue,
);

export const getshowMapLink = createSelector(
	[ venueBlockSelector ],
	( block ) => block.showMapLink,
);

export const getshowMap = createSelector(
	[ venueBlockSelector ],
	( block ) => block.showMap,
);
