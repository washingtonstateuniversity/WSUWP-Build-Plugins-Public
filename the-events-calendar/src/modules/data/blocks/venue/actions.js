/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';

export const setVenue = ( id ) => ( {
	type: types.SET_VENUE,
	payload: {
		venue: id,
	},
} );

export const removeVenue = () => ( {
	type: types.SET_VENUE,
	payload: {
		venue: undefined,
	},
} );

export const toggleVenueMap = () => ( { type: types.TOGGLE_VENUE_MAP } );

export const setShowMap = ( showMap ) => ( {
	type: types.SET_VENUE_MAP,
	payload: {
		showMap,
	},
} );

export const toggleVenueMapLink = () => ( { type: types.TOGGLE_VENUE_MAP_LINK } );

export const setShowMapLink = ( showMapLink ) => ( {
	type: types.SET_VENUE_MAP_LINK,
	payload: {
		showMapLink,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

