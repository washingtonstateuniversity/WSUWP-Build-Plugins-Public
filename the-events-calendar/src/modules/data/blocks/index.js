/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
