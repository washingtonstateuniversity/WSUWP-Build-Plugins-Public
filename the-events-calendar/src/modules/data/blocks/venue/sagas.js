/**
 * External Dependencies
 */
import { put, all, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';

export function* setInitialState( action ) {
	const { get } = action.payload;
	const venue = get( 'venue' ) ? get( 'venue' ) : DEFAULT_STATE.venue;

	yield all( [
		put( actions.setVenue( venue ) ),
		put( actions.setShowMap( get( 'showMap', DEFAULT_STATE.showMap ) ) ),
		put( actions.setShowMapLink( get( 'showMapLink', DEFAULT_STATE.showMapLink ) ) ),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
