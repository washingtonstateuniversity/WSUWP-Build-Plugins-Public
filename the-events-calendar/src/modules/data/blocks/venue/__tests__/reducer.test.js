/**
 * Internal dependencies
 */
import reducer, { DEFAULT_STATE } from '@moderntribe/events/data/blocks/venue/reducer';
import { actions } from '@moderntribe/events/data/blocks/venue';

describe( '[STORE] - Venue reducer', () => {
	it( 'Should set the initial state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set the venue', () => {
		expect( reducer( DEFAULT_STATE, actions.setVenue( 99 ) ) ).toMatchSnapshot();
	} );

	it( 'Should remove the venue', () => {
		expect( reducer( DEFAULT_STATE, actions.removeVenue() ) ).toMatchSnapshot();
	} );

	it( 'Should set the showMap', () => {
		expect( reducer( DEFAULT_STATE, actions.setShowMap( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setShowMap( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the showMapLink', () => {
		expect( reducer( DEFAULT_STATE, actions.setShowMapLink( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setShowMapLink( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should toggle the venue map link', () => {
		expect( reducer( DEFAULT_STATE, actions.toggleVenueMapLink() ) ).toMatchSnapshot();
	} );

	it( 'Should toggle the venue map', () => {
		expect( reducer( DEFAULT_STATE, actions.toggleVenueMap() ) ).toMatchSnapshot();
	} );
} );
