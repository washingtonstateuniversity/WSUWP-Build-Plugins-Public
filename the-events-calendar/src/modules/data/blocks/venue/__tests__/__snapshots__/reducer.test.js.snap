// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Venue reducer Should remove the venue 1`] = `
Object {
  "showMap": true,
  "showMapLink": true,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should set the showMap 1`] = `
Object {
  "showMap": true,
  "showMapLink": true,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should set the showMap 2`] = `
Object {
  "showMap": false,
  "showMapLink": true,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should set the showMapLink 1`] = `
Object {
  "showMap": true,
  "showMapLink": true,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should set the showMapLink 2`] = `
Object {
  "showMap": true,
  "showMapLink": false,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should set the venue 1`] = `
Object {
  "showMap": true,
  "showMapLink": true,
  "venue": 99,
}
`;

exports[`[STORE] - Venue reducer Should toggle the venue map 1`] = `
Object {
  "showMap": false,
  "showMapLink": true,
  "venue": undefined,
}
`;

exports[`[STORE] - Venue reducer Should toggle the venue map link 1`] = `
Object {
  "showMap": true,
  "showMapLink": false,
  "venue": undefined,
}
`;
